File: mobile/src/app/models/perfumerecognitionresult.dto.ts
import { PerfumeRecognitionConfidence } from '../enums/perfumerecognitionconfidence.enum';

export interface PerfumeRecognitionResultDto {
  perfumeId: number;
  score: number;
  perfumeName: string;
  brandName: string;
  imageUrl?: string;
  confidence: PerfumeRecognitionConfidence;
}

File: backend/Core/DTOs/PerfumeRecognitionResultDto.cs
using Core.Enums;

namespace Core.DTOs;

public sealed class PerfumeRecognitionResultDto
{
    public int PerfumeId { get; set; }
    public float Score { get; set; }

    public string PerfumeName { get; set; } = null!;
    public string BrandName { get; set; } = null!;
    public string? ImageUrl { get; set; }

    public PerfumeRecognitionConfidenceEnum Confidence { get; set; }
}
